config/zephyr/Kconfig

```diff
@@ -95,6 +95,7 @@ config CHIP_DEVICE_HARDWARE_VERSION_STRING
 	  hardware version string is exposed as an attribute of the Basic
 	  Information cluster.
 
+if "$(APPVERSION)"=""
```

Comment:
Could you update the PR description with the desired behavior? Do we want the VERSION file to take precedence over CHIP_DEVICE_SOFTWARE_VERSION? If so, what should a customer do in case they already use a different versioning scheme than ${APPVERSION}?

Reply:
Good point, thanks. User can still use CHIP_DEVICE_SOFTWARE_VERSION, we wanted to keep backward compability. Only when VERSION file is created the kconfig are disabled.